dwa_controller:
  ros__parameters:
    max_speed: 0.8
    min_speed: 0.0
    max_yaw_rate: 1.0
    max_accel: 0.5
    max_dyaw_rate: 2.0
    v_reso: 0.01
    yaw_reso: 0.03       # 0.01だと計算が重いので0.05に戻し、正規化で精度をカバー
    dt: 0.1
    predict_time: 1.0
    
    # 正規化後のスコアに対する重み (大きいほど重視)
    # heading: ゴールを向く力
    # dist: 障害物を避ける力
    # velocity: 速く進む力
    
    heading_score_gain: 0.04  # ゴールへの指向性 (振動するときは小さく)
    dist_score_gain: 0.1    # 障害物回避 (安全第一なら大きく)
    velocity_score_gain: 0.2 # 移動速度 (直進安定性に寄与)
    
    robot_radius: 0.2
    goal_tolerance: 0.5
    
    turn_p_gain: 0.7  # Pゲイン
    turn_d_gain: 1.0  # Dゲイン
